components/Main/index.js

```diff
@@ -1,22 +1,155 @@
-import { getLocalStorage } from "../../helpers/localStorage.js";
-import { getProjectFromLocalStorageById } from "../../services/project.js";
+import {
+  getLocalStorage,
+  setLocalStorage,
+} from "../../helpers/localStorage.js";
+import {
+  getProjectById,
+  deleteProject,
+  updateProjects,
+} from "../../services/project.js";
+import { previewProjectsNameOnTheSidebar } from "../Sidebar/index.js";
+import { renderCurrentProject } from "../Sidebar/index.js";
 let proName = document.querySelector(".project-name");
+//  AA!243232
+const threeDotsIcon = document.querySelector(".three-dots-icon");
+const redDeleteIcon = document.querySelector(".red-delete-icon");
+const greenTrueIcon = document.querySelector(".green-true-icon");
+let wantedProject;
 export const viewProjectNameOnMainPage = () => {
-  let wantedProject = getProjectFromLocalStorageById(
-    +localStorage.getItem("currentProjectIdFromSidebar")
+  wantedProject = getProjectById(
+    getLocalStorage("currentProjectIdFromSidebar")
   );
+  if (!wantedProject) {
+    proName.textContent = "NoProjectsYet";
+    threeDotsIcon.classList.add("hide");
+  } else {
+    proName.textContent = wantedProject.projectName;
+    threeDotsIcon.classList.remove("hide");
+  }
+};
+const actionMenuContainer = document.querySelector(".actions-container");
+export const deleteProjectUiAction = (deletedId) => {
+  let projects = getLocalStorage("projects");
+  const project = projects.find((item) => item.ProjectId == deletedId);
+  let currantIndex = projects.indexOf(project);
+
+  if (currantIndex == 0 && projects.length > 1) {
+  } else {
+    currantIndex--;
+  }
+
+  projects = deleteProject(deletedId);
+
+  const newId = currantIndex < 0 ? null : projects[currantIndex].ProjectId;
+
+  renderCurrentProject(newId);
+  previewProjectsNameOnTheSidebar();
+
+  const liElement = document.getElementById(newId);
+
+  if (liElement) {
+    liElement.classList.add("selected-project");
+  }
+};
+
+let actionForm = document.forms["actionForm"]["edit-project-name-input"];
+
+const editProjectUiAction = () => {
+  //copied code
+  let EditProjectNameInput = actionForm;
+  wantedProject = getProjectById(
+    getLocalStorage("currentProjectIdFromSidebar")
+  );
+  EditProjectNameInput.value = wantedProject.projectName;
+  function isValidString(input) {
+    const regex = /^[a-zA-Z0-9 ]*$/;
+    return regex.test(input);
+  }
+  let errorMSG = document.createElement("div");
+
+  errorMSG.style.color = "red";
+  errorMSG.style.fontSize = "11px";
+  errorMSG.style.fontWeight = "500";
+  errorMSG.style.position = "absolute";
+  errorMSG.style.left = " 305px";
+  errorMSG.style.top = " 80px";
```

Comment:
1. Move isValidString to a shared utilities file if used in multiple places.
```js
// utils.js
export function isValidString(input) {
    const validPattern = /^[a-zA-Z0-9 ]*$/;
    return validPattern.test(input);
}
```
2. Inline styles on errorMSG could instead be defined in CSS for easier maintenance.

```css
// CSS
.error-message {
    color: red;
    font-size: 11px;
    font-weight: 500;
    position: absolute;
    left: 305px;
    top: 80px;
}
```
```js
// JavaScript
const errorMsg = document.createElement("div");
errorMsg.className = "error-message";
```